### DISCUSSION TITLE
Mirth Connect Administrator Launcher: "Error reading Java home directory: ../../jre"

### DISCUSSION THREAD

**Discussion by jonathanleistikobcx** [nextgenhealthcare/connect#5026] [Q&A]
Howdy,

I suspect this is related to the issue (https://forums.mirthproject.io/forum/mirth-connect/support/18561-launcher-error-error-reading-java-home-directory-plugins-jre-bundle-c) on the prior forum.

When I use Mirth Connect Administrator Launcher to open a connection (click the Launch button to the right of the address), it gets to "All done! Launching Administrator," pauses for a moment, then displays the following error: 

```
Error Downloading / Verifying Codebase

Error reading Java home directory: ../../jre
java.lang.Exception: Error reading Java home directory: ../../jre
	at com.mirth.connect.client.launcher.O.a(SourceFile:125)
	at com.mirth.connect.client.launcher.MirthClientLauncher.run(SourceFile:2639)
	at java.lang.Thread.run(Unknown Source)
```
<img width="417" alt="Screen Shot 2022-02-15 at 11 07 53 AM" src="https://user-images.githubusercontent.com/52299385/154112552-fd2f5cea-8e63-4dd0-b30d-a82903558735.png">

Per info from the old article, I'm getting ready to dig through my bash profile and see if I can find a cause. I'm posting here in case someone has insight or a useful suggestion. Thanks!

New info 2022-02-16: 
Mini follow-up: I haven't had time to properly troubleshoot this, though I periodically tried it again. The issue persisted after a reboot, but when I signed out, then signed back in (Windows), I got it to work. Did signing out and back in "fix" it? I doubt it, but I'm jotting it here because extra info is almost always helpful.

**Comment by jonathanleistikobcx** [comment]
Mini update: Issue seems to have disappeared, but now the launcher intermittently opens as a window that's all white with no text or other interface elements. Hovering over it causes the cursor to change from pointer to insertion cursor in areas you'd expect a text field, but clicking where you know a button is does not make it attempt to start a connection. Restarting seems to resolve this.

...so the situation has gotten weird.

**Comment by mlarcelet-cmd** [comment]
I experience the exact same problem since I upgraded my launcher to v1.5.0 : error message "Error Downloading / Verifying Codebase

Error reading Java home directory: ../../jre
java.lang.Exception: Error reading Java home directory: ../../jre
	at com.mirth.connect.client.launcher.O.a(SourceFile:125)
	at com.mirth.connect.client.launcher.MirthClientLauncher.run(SourceFile:2639)
	at java.lang.Thread.run(Unknown Source)" when first try to connect to any mirthconnect server. The second attempt is OK.
